### PR TITLE
[dhcp-relay] Validate route to DHCP server

### PR DESCRIPTION
**Pull request by tahmed-dev**
### Description of PR
Adding fixture to verify that routes to DHCP server are present
on DUT before starting the test. This will exclude one problem when
debugging DHCP relay test issues.

signed-off-by: Tamer Ahmed <tamer.ahmed@microsoft.com>

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
Help debugging DHC relay test cases errors

#### How did you do it?
Provided new fixture to verify route

#### How did you verify/test it?
```
py.test --inventory veos.vtb --host-pattern all --user admin -vvv --show-capture stdout --testbed vms-kvm-t0 --testbed_file vtestbed.csv --disable_loganalyzer --junitxml=tr.xml dhcp_relay/test_dhcp_relay.py
```

### CODE REVIEW

File: tests/dhcp_relay/test_dhcp_relay.py [hunk 1]
```diff
@@ -96,8 +96,21 @@ def dut_dhcp_relay_data(duthost, ptfhost, testbed):
...
+def validate_dut_routes_exist(duthost, dut_dhcp_relay_data):
+    """Fixture to valid a route to each DHCP server exist
+    """
+    dhcp_servers = set()
+    for dhcp_relay in dut_dhcp_relay_data:
+        dhcp_servers |= set(dhcp_relay['downlink_vlan_iface']['dhcp_server_addrs'])
+
+    for dhcp_server in dhcp_servers:
+        result = duthost.shell("ip route get {0}".format(dhcp_server))
+        assert result["rc"] == 0 and "{0} via".format(dhcp_server) in result["stdout"], \
```
Comment (jleveque) [inline tests/dhcp_relay/test_dhcp_relay.py:109]: Is `assert` working properly now, or should we use `pytesl.fail()` or Ying's new `pytest_assert`?

Comment (tahmed-dev) [reply tests/dhcp_relay/test_dhcp_relay.py:109]: Yes, I had no issues with assert. I have seen the assert with no locals variable message being triggered.

Comment (tahmed-dev) [reply tests/dhcp_relay/test_dhcp_relay.py:109]: here is a sample output of the assert (please note condition inversion in order to trigger assert)
```
duthost = <common.devices.SonicHost object at 0x7fcae13f5490>
dut_dhcp_relay_data = [{'client_iface': {'alias': u'fortyGigE0/52', 'name': u'Ethernet52', 'port_idx': 13}, 'downlink_vlan_iface': {'addr': ...['PortChannel0002', 'PortChannel0001', 'PortChannel0004', 'PortChannel0003'], 'uplink_port_indices': [29, 28, 31, 30]}]

    @pytest.fixture(scope="module")
    def validate_dut_routes_exist(duthost, dut_dhcp_relay_data):
        """Fixture to valid a route to each DHCP server exist
        """
        dhcp_servers = set()
        for dhcp_relay in dut_dhcp_relay_data:
            dhcp_servers |= set(dhcp_relay['downlink_vlan_iface']['dhcp_server_addrs'])
    
        for dhcp_server in dhcp_servers:
            result = duthost.shell("ip route get {0}".format(dhcp_server))
>           assert result["rc"] != 0 and "{0} via".format(dhcp_server) in result["stdout"], \
                "Failed to find route to DHCP server '{0}' with error '{1}".format(dhcp_server, result["stderr"])
E           AssertionError: Failed to find route to DHCP server '192.0.0.4' with error '
E           assert (0 != 0)

dhcp_relay/test_dhcp_relay.py:109: AssertionError
```

File: tests/dhcp_relay/test_dhcp_relay.py [hunk 2]
```diff
@@ -96,8 +96,21 @@ def dut_dhcp_relay_data(duthost, ptfhost, testbed):
...
+@pytest.fixture(scope="module")
+def validate_dut_routes_exist(duthost, dut_dhcp_relay_data):
+    """Fixture to valid a route to each DHCP server exist
+    """
+    dhcp_servers = set()
+    for dhcp_relay in dut_dhcp_relay_data:
+        dhcp_servers |= set(dhcp_relay['downlink_vlan_iface']['dhcp_server_addrs'])
+
+    for dhcp_server in dhcp_servers:
+        result = duthost.shell("ip route get {0}".format(dhcp_server))
```
Comment (lguohan) [inline tests/dhcp_relay/test_dhcp_relay.py:108]: can you reuse this function? duthost.get_ip_route_info(ipaddress.ip_address(u"0.0.0.0"))

example here.

https://github.com/Azure/sonic-mgmt/blob/master/tests/route/test_default_route.py#L31

Comment (tahmed-dev) [reply tests/dhcp_relay/test_dhcp_relay.py]: Done

**Pull request merged**